Hestia/app/src/main/java/hestia/UI/DeviceListFragment.java

```diff
@@ -23,13 +27,14 @@
  * @see DeviceListActivity
  */
 public class DeviceListFragment extends Fragment implements DevicesChangeListener{
-
-
+    private SwipeRefreshLayout swipeRefreshLayout;
     private ExpandableListAdapter listAdapter;
     private ExpandableListView expListView;
     private ArrayList<ArrayList<DeviceBar>> listDataChild;
     private BackendInteractor backendInteractor =  BackendInteractor.getInstance();
-    private FloatingActionButton fab;
+    private FloatingActionButton floatingActionButton;
+    private final static String TAG = "DeviceListFragment";
+    private Activity surroundingActivity;
 
     /**
```

Comment:
Some of your methods and classes are still quite large. It can be decreased by not always defining anonymous classes within your methods (just as an example).